js/state/siteUtil.js

```diff
@@ -334,3 +335,22 @@ module.exports.clearSitesWithoutTags = function (sites) {
 module.exports.hasNoTagSites = function (sites) {
   return sites.findIndex((site) => !site.get('tags') || site.get('tags').size === 0) !== -1
 }
+
+/**
+ * Gets a site origin (scheme + hostname + port) from a URL or null if not
+ * available.
+ * @param {string} location
+ * @return {string?}
+ */
+module.exports.getOrigin = function (location) {
+  // Returns scheme + hostname + port
+  if (typeof location !== 'string') {
+    return null
+  }
+  let parsed = urlParse(location)
+  if (parsed.host) {
+    return location.split(parsed.host)[0] + parsed.host
```

Comment:
minor bug here where if a hostname is the same as a protocol name then this won't work correctly.   Maybe I could have a computer on my network called http for example.  Far fetched example but could happen.